lib/arel_extensions/null_functions.rb

```diff
@@ -19,5 +19,15 @@ def coalesce *args
       args.unshift(self)
       ArelExtensions::Nodes::Coalesce.new args
     end
+
+    def coalesce_blank *args
+      res = Arel.when(self.cast(:string).present).then(self)
+      args[0...-1].each do |a|
+        val = a.is_a?(Arel::Nodes::Node) ? a : Arel.quoted(a)
+        res = res.when(val.present).then(a)
+      end
+      res.else(args[-1])
```

Comment:
You need the assignment here.  You are returning without the `else` taken into account.

Could you enhance the test case to catch this failure?

And btw, please enrich the commit message of the commit about Case to explain why you made that change.

Reply:
This comment was wrong, what could be seen from reading the test suite, since it covers that case.  It turns out that `when` is "sticky".  It's always updated in place.
```ruby
q = Arel.when(0).then(1)
puts q.to_sql
# => CASE WHEN 0 THEN 1 END
q = q.when(Arel.quoted(2)).then(3)
puts q.to_sql
# => CASE WHEN 0 THEN 1 WHEN 2 THEN 3 END
q.else(4)
puts q.to_sql
# => CASE WHEN 0 THEN 1 WHEN 2 THEN 3 ELSE 4 END
```

That being said, it will be nicer with the assignment anyway.

Reply:
Done, and done.